### PR TITLE
update kcl.mod package.edition with the kcl compiler version

### PR DESCRIPTION
**Pull request by AkashKumar7902**
Solves: #186

Signed-off-by: Akash Kumar <meakash7902@gmail.com>

### CODE REVIEW

**Comment by AkashKumar7902** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by AkashKumar7902** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by AkashKumar7902** [comment]
@zong-zhe Please take a look!

File: pkg/client/client.go [hunk 1]
```diff
@@ -325,6 +325,11 @@ func (c *KpmClient) Compile(kclPkg *pkg.KclPkg, kclvmCompiler *runner.Compiler)
 		kclvmCompiler.AddDepPath(dName, dPath)
 	}
 
+	kclPkg.UpdatePkgEdition(kclvmCompiler.GetKclVersion())
+	if err := kclPkg.ModFile.StoreModFile(); err != nil {
```
Comment (Peefy) [inline pkg/client/client.go:329]: Remove the function `kclPkg.ModFile.StoreModFile` calling.

Comment (AkashKumar7902) [reply pkg/client/client.go]: done! 
I have made changes according to https://github.com/kcl-lang/kpm/issues/186#issuecomment-1950987112

**Comment by AkashKumar7902** [comment]
@Peefy can you please rerun "unit test" pipeline ? looks like a flaky github action run

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**